homeassistant/components/google/__init__.py

```diff
@@ -130,6 +142,20 @@
 )
 
 
+def get_scope(conf):
+    """Gets the google calendar scope based on the defined access level."""
+
+    scopes = ""
```

Comment:
I was looking at some other examples of using enums for config in home assistant and came across what `zha` does with the `RadioType` enum: https://github.com/home-assistant/core/blob/470f2dd73f9f476cf5e7ba58885af2aea43169d5/homeassistant/components/zha/core/const.py#L233

Some of this logic can go away if the access is defined as an enum with the scope baked in, something like:

```
class FeatureAccess(Enum):
   read_only = ("https://www.googleapis.com/auth/calendar.readonly")
   readwrite = ("https://www.googleapis.com/auth/calendar")

  def __init__(self, scope: str) -> None:
      """..."""
      self._scope = scope

  @property
  def scope(self) -> str:
      """..."""
      return self._scope
```

Then I think you can just get the enum value from the config and pull the scope out of it all inline.

Reply:
Didn't realise this was a thing. Updated.